zebrad/tests/acceptance.rs

```diff
@@ -461,3 +464,119 @@ fn valid_generated_config(command: &str, expected_output: &str) -> Result<()> {
+    std::thread::sleep(Duration::from_secs(1));
+
+    // Create an http client
+    let client = Client::new();
+
+    // Test metrics endpoint
+    let res = client.get(Uri::from_static(url)).await?;
+    assert!(res.status().is_success());
+    let body = hyper::body::to_bytes(res).await?;
+    assert!(std::str::from_utf8(&body)
+        .unwrap()
+        .contains("metrics snapshot"));
+
+    child.kill()?;
+
+    let output = child.wait_with_output()?;
+    let output = output.assert_failure()?;
+
+    // Make sure metrics was started
+    output.stdout_contains(format!(r"Initializing metrics endpoint at {}", endpoint).as_str())?;
+
+    // Make sure the command was killed
+    output.assert_was_killed()?;
+
+    Ok(())
+}
+
+#[tokio::test]
+#[ignore]
+async fn tracing_endpoint() -> Result<()> {
+    use hyper::{Body, Client, Request, Uri};
+
+    zebra_test::init();
+
+    // [Note on port conflict](#Note on port conflict)
+    let endpoint = "127.0.0.1:50002";
+    let url_default = "http://127.0.0.1:50002";
+    let url_filter = "http://127.0.0.1:50002/filter";
+
+    // Write a configuration that has tracing endpoint_addr option set
+    let mut config = default_test_config()?;
+    config.tracing.endpoint_addr = Some(endpoint.parse().unwrap());
+
+    let dir = TempDir::new("zebrad_tests")?;
+    fs::File::create(dir.path().join("zebrad.toml"))?
+        .write_all(toml::to_string(&config)?.as_bytes())?;
+
+    let tempdir = dir.path().to_path_buf();
+
+    let mut child = get_child(&["start"], &tempdir)?;
+
+    // Run the program for a second before testing the endpoint
+    std::thread::sleep(Duration::from_secs(1));
+
+    // Create an http client
+    let client = Client::new();
+
+    // Test tracing endpoint
+    let res = client.get(Uri::from_static(url_default)).await?;
+    assert!(res.status().is_success());
+    let body = hyper::body::to_bytes(res).await?;
+    assert!(std::str::from_utf8(&body).unwrap().contains(
+        "This HTTP endpoint allows dynamic control of the filter applied to\ntracing events."
+    ));
+
+    // Set a filter and make sure it was changed
+    let request = Request::post(url_filter)
+        .body(Body::from("zebrad=debug"))
+        .unwrap();
+    let _post = client.request(request).await?;
+
+    let tracing_res = client.get(Uri::from_static(url_filter)).await?;
+    assert!(tracing_res.status().is_success());
+    let tracing_body = hyper::body::to_bytes(tracing_res).await?;
+    assert!(std::str::from_utf8(&tracing_body)
+        .unwrap()
+        .contains("zebrad=debug"));
+
+    child.kill()?;
+
+    let output = child.wait_with_output()?;
+    let output = output.assert_failure()?;
+
+    // Make sure tracing endpoint was started
+    output.stdout_contains(format!(r"Initializing tracing endpoint at {}", endpoint).as_str())?;
+    // Todo: Match some trace level messages from output
+
+    // Make sure the command was killed
+    output.assert_was_killed()?;
```

Comment:
Let's add a useful hint to the error message.

We'll need to use eyre::WrapErr to use this function.

```suggestion
    output.assert_was_killed().wrap_err("Possible port conflict. Are there other acceptance tests running?")?;
```

Reply:
https://github.com/ZcashFoundation/zebra/pull/1000/commits/4e6bca466a78290c4a4d9e2906521cdcaf66ca54